File: lib/models/quick_chat_dto.dart
class QuickChatDto {
  final String message;
  final String? subject;
  final String? grade;

  QuickChatDto({
    required this.message,
    this.subject,
    this.grade,
  });

  Map<String, dynamic> toJson() {
    return {
      'message': message,
      if (subject != null) 'subject': subject,
      if (grade != null) 'grade': grade,
    };
  }

  factory QuickChatDto.fromJson(Map<String, dynamic> json) {
    return QuickChatDto(
      message: json['message'] as String,
      subject: json['subject'] as String?,
      grade: json['grade'] as String?,
    );
  }
}

File: backend/src/smart-tools/dto/quick-chat.dto.ts
import { IsString, IsNotEmpty, Length, IsOptional } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';

export class QuickChatDto {
  @ApiProperty({ description: 'Mesaj içeriği', maxLength: 1000, example: 'Fotosentez nasıl çalışır?' })
  @IsString()
  @IsNotEmpty()
  @Length(1, 1000)
  message!: string;

  @ApiProperty({ description: 'Konu/alan', required: false, example: 'Biyoloji' })
  @IsString()
  @IsOptional()
  subject?: string;

  @ApiProperty({ description: 'Sınıf seviyesi', required: false, example: '11' })
  @IsString()
  @IsOptional()
  grade?: string;
}
